data-service/src/main/java/uk/gov/gchq/palisade/service/data/DataApplication.java

```diff
@@ -50,36 +57,49 @@
 public class DataApplication {
     private static final Logger LOGGER = LoggerFactory.getLogger(DataApplication.class);
 
-    private final DataReader dataReader;
     private final AuditMessageService auditMessageService;
-
+    private final DataReader dataReader;
     private final StdSerialiserConfiguration serialiserConfiguration;
+
     private final RunnableGraph<Sink<TokenMessagePair, NotUsed>> runner;
+    private final Set<ActorSystem> actorSystems;
+    private final AkkaHttpServer server;
+    private final ActorSystem system;
     private final Materializer materialiser;
-    private CompletableFuture<?> runnerThread = new CompletableFuture<>();
+    private final Executor executor;
+    private final Set<CompletableFuture<?>> runnerThreads = new HashSet<>();
 
     /**
-     * Constructor for {@code DataApplication}.
+     * Autowire Akka objects in constructor for application ready event
      *
-     * @param dataReader              a reader for retrieving the request resources.
-     * @param auditMessageService     service for sending audit success and error messages to the Audit Service
-     * @param serialiserConfiguration a configuration and initialising the {@link DataReader}
-     * @param runner                  runnable graphs for sending messages on the Kafka stream
-     * @param materialiser            the Akka {@link Materializer} responsible for turning a stream blueprint into a
-     *                                running stream.
+     * @param auditMessageService     instance of the auditMessageService to connect with the sink from the runner
+     * @param dataReader              reader to 'prepopulate' with serialisers
```

Comment:
I think  the re-wording of the JavaDoc could be revisited.   The description of the dataReader is a good example.  The original phrase "a reader for retrieving the request resources" is not great, but "reader to 'prepopulate' with seralisers" does not tell us what a dataReader is.  I would vote for more on the concept and less on the technical.

Reply:
This shouldn't be the place to describe what a `DataReader` is, that's appropriate for the javadoc on the `DataReader` class itself. Instead, we document what we do with it and why it's used here. It does not get used in this context for reading resources, that would be more appropriate for the `DataService` javadoc